### PR TITLE
Update submodule to latest `release-branch.go1.20` in `microsoft/release-branch.go1.20`

### PR DESCRIPTION
**Pull request by ghost**
Hi! I'm a bot, and this is an automatically generated upstream sync PR. 🔃

After submitting the PR, I will attempt to enable auto-merge in the "merge commit" configuration.

For more information, visit [sync documentation in microsoft/go-infra](https://github.com/microsoft/go-infra/tree/main/docs/automation/sync.md).

### CODE REVIEW

**Review by ghost — APPROVED** [review]
Thanks! Auto-approving.

**Comment by dagood** [comment]
RSA test build failed after the initial minimal fixup.

File: patches/0006-Add-CNG-crypto-backend.patch [hunk 1]
```diff
@@ -692,26 +692,26 @@ index 512e4a4d6616e5..284db01c221d93 100644
...
-index 379fb06526ccc7..d844cc4b8d749e 100644
+index fb3f821e76a6d4..6972ccbf2fcbde 100644
 --- a/src/crypto/rsa/rsa_test.go
 +++ b/src/crypto/rsa/rsa_test.go
 @@ -8,6 +8,7 @@ import (
  	"bufio"
  	"bytes"
  	"crypto"
 +	boring "crypto/internal/backend"
+ 	"crypto/internal/boring"
```
Comment (qmuntal) [inline patches/0006-Add-CNG-crypto-backend.patch]: This line should be deleted and the previous line addition moved to Patch-003

**Review by qmuntal — DISMISSED** [review]

File: patches/0006-Add-CNG-crypto-backend.patch (hunk 1 above)
Comment (qmuntal) [reply patches/0006-Add-CNG-crypto-backend.patch]: See what I did in the upstream sync PR: https://github.com/microsoft/go/commit/c244632c6df4af681bf5d0f2c14f5d3014a505d7

**Pull request merged**